### DISCUSSION TITLE
Feature Request: ADLS2 support in HTTPFS extension

### DISCUSSION THREAD

**Discussion by ttomasz** [duckdb/duckdb#5574] [Ideas]
Why do you want this feature?

HTTPFS extension supports S3 but it would be nice if it supported also Azure Storage since as far as I can tell Azure does not have API compatible with S3 like GCP does.

Azure docs: https://learn.microsoft.com/en-us/rest/api/storageservices/data-lake-storage-gen2

**Comment by Mause** [comment]
It looks like the official way to add s3 API support is to use a proxy: https://devblogs.microsoft.com/cse/2016/05/22/access-azure-blob-storage-from-your-apps-using-s3-api/

**Reply by sdebruyn** [reply]
The proxy comes with extra setup & performance hit

**Reply by Mause** [reply]
Since this question was asked, we've added fsspec support, which will probably help. Not sure how comparable the performance is though

**Reply by sdebruyn** [reply]
That does indeed help! 😁 It would be nice to have this natively just like we have S3 though
